doc/conf.py

```diff
@@ -112,14 +112,12 @@ def setup(app):
     "sphinx_gallery.gen_gallery",
 ]
 
-# private-member docs are generally not great to link to in high-level implementation
-# documentation because the implementation may change rapidly. Prefer putting info in
-# public entities. We  render docs with private-members shown, however, because there
-# are important implementation details in them in many cases.
```

Comment:
I do love this PR. This one line is the one thing I think might warrant more discussion. In the plugins in particular there are in some cases vast docstrings of math that describe an implementation that I think are really valuable, and this will hide them from the rendered docs. Should we maybe just move all those out of private docstrings to handle this?

Reply:
Yeah, this is worth a discussion.

1. This PR doesn't apply to plugins, right? If a plugin has important data hidden inside a private method docstring, then that plugin could decide to document it's private members.
2. You essentially hit the nail on the head. If something is _so_ important, we need to document it, than it shouldn't be in a private method. That method should be public. OR, of course, we could just document that one thing elsewhere, say in the class or module docstring. 

My main goal here is to start thinking about the "API" of our code. Sure, people can always "read all the source code" to understand what a program does. But, typically, more than 50% of a codebase is just implementation details. And what makes life easier for a user is to document the minimal interface they need to understand to interact with the code.

Maybe in our code base, we should add important notes on scientific data in class/module docstrings, because we want to document the SCIENCE.  I love that.  That can help a user way more than documenting 40 private methods on a class that the user doesn't ever need to know about.

Reply:
Good points all around. Plugins can indeed still choose to show private docstrings. And yeah maybe class docstrings are generally a better place for that level of implementation detail. And if the plugin author wants private, they can keep them turned on. 

We may want to scan through for particularly useful private docstrings in the framework after landing this just in case.